node/libs/concurrency/src/sync/prunable_mpsc/mod.rs

```diff
@@ -14,15 +14,30 @@ use std::{collections::VecDeque, fmt, sync::Arc};
 #[cfg(test)]
 mod tests;
 
+/// Possible results for the filter function
+#[derive(Debug, PartialEq)]
+pub enum SelectionFunctionResult {
+    /// Keep both values.
+    Keep,
+    /// Discard old value.
+    DiscardOld,
+    /// Discard new value.
+    DiscardNew,
+}
+
 /// Creates a channel and returns the [`Sender`] and [`Receiver`] handles.
 /// All values sent on [`Sender`] will become available on [`Receiver`] in the same order as it was sent,
 /// unless will be pruned before received.
 ///
 /// * [`T`]: The type of data that will be sent through the channel.
-/// * [`pruning_predicate`]: A function that determines whether an unreceived, pending value in the buffer (represented by the first `T`) should be pruned
-/// based on a newly sent value (represented by the second `T`).
+/// * [`filter_function`]: A function that checks the newly sent value and avoids adding to the queue if it returns false
```

Comment:
comment outdated

Reply:
nit: actually DiscardOld is not meaningful for filter_function, so perhaps bool was a better choice there?

Reply:
Yeah, that was my first version, but I found the code clearer defining both functions returning a `SelectionFunctionResult`  (like [here](https://github.com/matter-labs/era-consensus/blob/5d200d81073708f06a4a5911ab2c48c12739c103/node/actors/bft/src/replica/state_machine.rs#L173-L209))

Reply:
I debated myself on what version is better. I'm open for suggestions.

Reply:
well, you can have 2 enums.

Reply:
Going for the boolean version. It's cleaner to define a predicate.